### ISSUE TITLE
Syntax error for gen_vals

### ISSUE THREAD

**Issue by artembosten** [PaulBatchelor/Sporth#42]
I have a very simple file:
_signal FTABLE gen_vals
44100 metro 0 _signal tseq

I get a syntax error lin1 char9 which is the quotation mark at the start of the table values.  I don't see the error but the table is very large, could that be the issue?

**Comment by PaulBatchelor** [comment]
it's possible. is there any chance you can upload the file itself?

**Comment by artembosten** [comment]
I dont have the file on this device but I can do that tomorrow.  Could you clarify something for me?  generators (new to sporth, not sure what to call them yet) such as sine push to the stack every iteration, but if I set the trigger to tseq to 1 a stack overflow occurs.  Is it that sine honors sample rate and tseq doesnt?

**Comment by artembosten** [comment]
Here's the file as a txt:
[noteArray.txt](https://github.com/PaulBatchelor/Sporth/files/3051789/noteArray.txt)

**Comment by PaulBatchelor** [comment]
Wowee! that's a huge string array to put into `gen_vals`. It's not really built to handle strings of that size. I also noticed that you are setting the `metro` generator to be 44100, which is the default sampling rate. A very strange value to use, since you might as well put in "1".

What are you trying to do? If I didn't know any better, I would say you are trying to playback an audio sample. It really looks like there should be a better way to do what you are trying to do (tabread, nsmp, slice, etc).

To answer your question: everything is an signal in Sporth. Both tseq and sine are audio rate signals. Not exactly sure why tseq bugs out when the mode is changed to "1".

**Comment by artembosten** [comment]
For my code I did just read a wav file.  This popped up like you said when I plugged in a short audio file out of curiosity.  The size is less than the max allowed in csound - what's the max table size in sporth?  The overflow with 1 must be the same issue then.

**Comment by artembosten** [comment]
I think I misunderstood you.  If its the parser max table size is the same?

**Comment by PaulBatchelor** [comment]
The maximum size of an ftable is determined by =size_t=, which varies from system to system. The size correlates to the number of samples, not in bytes:

https://github.com/PaulBatchelor/Soundpipe/blob/35d856fcf8a0d85c285818554f569b592d066d10/h/ftbl.h#L5

I ran a version of your file in gdb, and got a segfault happening with the function reading the lines in the function `sporth_getline`:

https://github.com/PaulBatchelor/Sporth/blob/dev/parse.c#L287

This is a very simple getline implementation, so it wouldn't surprise me if things start to break when you try to read lines that are literally megabytes in size. You very well could be the first person to really test the limits of the parser in this way!

> For my code I did just read a wav file.

To be clear, you are technically reading a string generated *from* a wav file via `gen_vals`, which is different that reading the wav file. The `gen_vals` f-table function was not designed to be used to embed audio files like this. Is there any reason why you aren't using the "loadfile" ugen to load your audio file? It'll be a fraction of the size, and you won't have any data loss that would have been caused by the float to string conversion.
